extend the use of tagged-bytes to identifiers (#188)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

{::include cddl/digest.cddl}
~~~

### Tagged Bytes Type {#sec-common-tagged-bytes}

An opaque, variable-length byte string.

> It can be used in different contexts: as an instance, class or group identifier in an `environment-map`; as a raw value measurement in a `measurement-values-map`.

Reviewer: ```suggestion
It can be used in different contexts: as an instance, a class or a group identifier in an `environment-map`; as a raw value measurement in a `measurement-values-map`.
```

Reviewer 2: Agreed to omit this suggestion.

{::include cddl/digest.cddl}
~~~

### Tagged Bytes Type {#sec-common-tagged-bytes}

An opaque, variable-length byte string.
It can be used in different contexts: as an instance, class or group identifier in an `environment-map`; as a raw value measurement in a `measurement-values-map`.
Its semantics is defined by the context in which it is found, and by the overarching CoRIM profile.

> When used as an identifier the responsible allocator entity SHOULD ensure uniqueness.

Reviewer 3: the scope of uniqueness should be defined.

Author: You mean something like:

> _"[...] ensure uniqueness within the scope of the usage scope."_

Reviewer 3: > You mean something like:
> 
> > _"[...] ensure uniqueness within the usage scope."_

OK

Author: cool, see [9757398](https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/188/commits/9757398627525052f2b0bf3d2dd2c36affd790a8)

> When used as an identifier the responsible allocator entity SHOULD ensure uniqueness.

Author: ```suggestion
When used as an identifier the responsible allocator entity SHOULD ensure uniqueness within the usage scope.
```

> When used as an identifier the responsible allocator entity SHOULD ensure uniqueness within the usage scope.

Reviewer 4: ```suggestion
When used as an identifier the responsible allocator entity SHOULD ensure global uniqueness.
```

Author: I don't think _global_ uniqueness is needed here.  It is sufficient that the tagged bytes unambiguously identify an entity within a given scope.  The same `560(h'01')` may refer to instance "1" or to class "1" without ambiguity.

Reviewer 3: The other examples of `instance-id` have implied global uniqueness properties.  The motivation for adding `bytes` was that some felt ueid might not be big enough to ensure global uniqueness. The verifier doesn't have additional context to know what scope to apply to disambiguate scope-A:bytes from scope-B:bytes. We shouldn't overload class properties in `instance-id` since `class-id` is intended accommodate class-based identifiers.

Reviewer 3: Discussion today resolved that 560 is to be interpreted within the CDDL context in which it occurs. This PR proposes 3 additional contexts from the current raw-value-group context. The description of the CDDL for each context should describe the expected properties.

In the case, of `environment-map` there are other issue #176 observes the lack of meaningful scoping discussion for each of the fields (class, instance, group). If we resolve #176, then this thread becomes mute.

Author: > Discussion today resolved that 560 is to be interpreted within the CDDL context in which it occurs.

Yes

> This PR proposes 3 additional contexts from the current raw-value-group context.

I don't understand this.

> The description of the CDDL for each context should describe the expected properties.

Yes

> In the case, of `environment-map` there are other issue #176 observes the lack of meaningful scoping discussion for each of the fields (class, instance, group). If we resolve #176, then this thread becomes mute.

WFM

that is an instance of the Attester.

The types defined for an instance identifier are CBOR tagged expressions of

> UEID, UUID, variable-length opaque byte string, or cryptographic key identifier.

Reviewer: Please either add a reference to tagged bytes or describe profile specific behaviour dependency here!

> UEID, UUID, variable-length opaque byte string, or cryptographic key identifier.

Reviewer: Similar entry is needed in the class identifier as well!

Author: see [6fee7e3](https://github.com/ietf-rats-wg/draft-ietf-rats-corim/pull/188/commits/6fee7e39d2eaa8cf792443ede914bfcf53d8b71a)
